content/posts/entendendo-o-dom-document-object-model.md

```diff
@@ -0,0 +1,251 @@
+---
+title: Entendendo o DOM (Document Object Model)
+author: Leonardo Maldonado
+type: post
+image: https://imgur.com/y05YIAB
+date: 2018-02-05
+excerpt: O DOM explicado de uma maneira fácil.
+categories:
+  - JavaScript
+  - HTML
+---
+
+
+#### Entendendo o DOM (Document Object Model)
+
+Você estudou bastante HTML, criou suas primeiras tags, aprendeu CSS, fez formulários bonitos, botões incríveis, páginas responsivas e começou mostrar para todo mundo como aquilo é incrível, mas chega uma hora que você deseja dar um passo a mais no seu aprendizado e se pergunta: ”Mas como eu posso dar movimento a minha página? Queria tanto que aquele botão por exemplo mudasse algo nela!”. É aí que entra o DOM, algo que constantemente você ouve falar por aí, mas não sabe explicar ao certo o que é.
+
+Sabe aquelas animações legais que você vê por aí e você fala “Uau, queria tanto fazer algo assim, mas como?”, elas são feitas manipulando o DOM (Document Object Model), e esse artigo vai te explicar do zero o que ele é e como manipulá-lo.
+
+##### Então, o que é o DOM?
+
+O DOM (Document Object Model) é uma interface que representa como os documentos HTML e XML são lidos pelo seu browser. Após o browser ler seu documento HTML, ele cria um objeto que faz uma representação estruturada do seu documento e define meios de como essa estrutura pode ser acessada. Nós podemos acessar e manipular o DOM com JavaScript, é a forma mais fácil e usada.
+
+##### Quais as vantagens do DOM?
+
+Com ele você tem infinitas possibilidades, você pode criar aplicações que atualizam os dados da página sem que seja necessário uma atualização. Pode também criar aplicações que são customizáveis pelo usuário, mudar o layout da página sem que seja necessário atualização. Arrastar, mover, excluir elementos. Ou seja, você tem infinitas possibilidades, milhares de coisas que você pode fazer manipulando o DOM, basta você usar sua criatividade.
+
+##### Como ele é representado pelo browser:
+
+https://imgur.com/lu4PMfw
+
+A estrutura que o DOM constrói a partir da leitura do seu documento HTML.
+
+Nessa imagem vemos a estrutura do DOM, suas marcações e como ele é montado pelo browser. Nessa base, temos 4 pontos importantes que você vai ver bastante daqui pra frente:
+
+1. **Document: **que como o nome diz, cuida de documentos HTML.
+2. **Elements: **são todas as tags que estão em arquivos HTML ou XML se transformam em elementos da árvore DOM.
+3. **Texts**: É o texto que vai entre os elementos. Todo o conteúdo das tags.
+4. **Attributes**: É a junção de todos atributos para um nó específico. No caso, o attribute _class=”hero”_ está apontando para o elemento _&lt;p&gt;_.
+
+##### Manipulando o DOM
+
+Agora vamos a parte mais legal de todas: manipular o DOM. Primeiramente, vamos criar um HTML como exemplo para mostrar como os métodos e os eventos funcionam.
+
+O HTML:
```

Comment:
Coloca todo o código entre:

crase crase crase 
código 
crase crase crase 

Se não entendeu, for encontrar uma forma para explicar melhor. Revisa todos os código do artigo.